skyllh/core/parameters.py

```diff
@@ -1346,6 +1346,186 @@ def round_to_upper_grid_point(self, value):
         return gp
 
 
+class IrregularParameterGrid(object):
+    """This class provides a data holder for a parameter that has a set of
+    discrete values on a grid. Thus, the parameter has a value grid.
+    This class represents a one-dimensional irregular grid.
+    """
+    @staticmethod
+    def from_BinningDefinition(binning):
+        """Creates a IrregularParameterGrid instance from a BinningDefinition instance.
+
+        Parameters
+        ----------
+        binning : BinningDefinition instance
+            The BinningDefinition instance that should be used to create the
+            ParameterGrid instance from.
+
+        Returns
+        -------
+        param_grid : instance of ParameterGrid
+            The created ParameterGrid instance.
+        """
+        return IrregularParameterGrid(
+            name=binning.name,
+            grid=binning.binedges,
+        )
+
+    def __init__(self, name, grid):
+        """Creates a new parameter grid.
+
+        Parameters
+        ----------
+        name : str
+            The name of the parameter.
+        grid : sequence of float
+            The sequence of float values defining the discrete grid values of
+            the parameter.
+        """
+        self.name = name
+        self.grid = grid
+
+    def __str__(self):
+        """Pretty string representation.
+        """
+        return '{:s} = {:s}'.format(
+            self._name, str(self._grid))
+
+    @property
+    def name(self):
+        """The name of the parameter.
+        """
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        if not isinstance(name, str):
+            raise TypeError(
+                'The name property must be of type str!')
+        self._name = name
+
+    @property
+    def grid(self):
+        """The numpy.ndarray with the grid values of the parameter.
+        """
+        return self._grid
+
+    @grid.setter
+    def grid(self, arr):
+        if not issequence(arr):
+            raise TypeError(
+                'The grid property must be a sequence!')
+        if not isinstance(arr, np.ndarray):
+            arr = np.array(arr, dtype=np.float64)
+        if arr.ndim != 1:
+            raise ValueError(
+                'The grid property must be a 1D numpy.ndarray!')
```

Comment:
Minor comment: couldn't it just accept any kind of 1D sequence and cat it to a numpy.ndarray?

Reply:
It does that already